crates/astria-composer/src/composer.rs

```diff
@@ -240,6 +240,7 @@ impl Composer {
         // run the collectors and executor
         spawn_geth_collectors(&mut geth_collectors, &mut geth_collector_tasks);
 
+        let _chain_id_result = executor.check_chain_ids().await.wrap_err("chain_id mismatch");
```

Comment:
This performs a check but there is no actual effect on the rest of the code. Effectively, this just performs a get and wraps the return value into a `Result`. You need to have some more logic (like returning early) to have an effect.

Having said that, move this into `Executor::run_until_stopped`. The composer task acts more like a collection of sub tasks, while the sub tasks should be self-contained (including verification of chain IDs).

Reply:
Resolved in f3c96ff